jfr-daemon/src/main/java/com/newrelic/jfr/daemon/JFRUploader.java

```diff
@@ -65,15 +66,19 @@ public JFRUploader(
 
   void handleFile(final Path dumpFile) {
     // At startup should we read all the events present? This could be multiple hours of recordings
-    if (lastSeen == null) {
-      lastSeen = nowProvider.get();
+    Instant eventsAfter = null;
+
+    if (lastSeen.isEmpty()) {
+      eventsAfter = Instant.EPOCH;
+    } else {
+      eventsAfter = lastSeen.get();
```

Comment:
Definitely underutilizing the `Optional` here.  Suggest this instead:

```suggestion
Instant eventsAfter = lastSeen.orElse(Instant.EPOCH);
```